src/crypto/CMakeLists.txt

```diff
@@ -87,45 +87,41 @@ target_link_libraries(bitcoin_crypto
 )
 
 if(HAVE_SSE41)
-  target_sources(bitcoin_crypto PRIVATE sha256_sse41.cpp)
-  set_property(SOURCE sha256_sse41.cpp
-    APPEND PROPERTY COMPILE_OPTIONS ${SSE41_CXXFLAGS}
-  )
-  target_compile_definitions(bitcoin_crypto
-    PRIVATE
-      ENABLE_SSE41
+  add_library(bitcoin_crypto_sse41 STATIC EXCLUDE_FROM_ALL
```

Comment:
Same comment as #171. Would OBJECT libraries be better here?

Reply:
Same as https://github.com/hebasto/bitcoin/pull/171#discussion_r1585405675 :)

Yes, it could. But CMake's developers recommend to use `STATIC` libraries if there is no compelling reasons for `OBJECT` ones. For example, from [Professional CMake: A Practical Guide 18th Edition](https://crascit.com/professional-cmake/release-notes/18th-edition/)

> Prefer defining static libraries over object libraries. Static libraries are simpler, have more
complete and robust support from earlier CMake versions and they are well understood by most
developers. Object libraries have their uses, but they are also less flexible than static libraries.

Reply:
> Object libraries have their uses, but they are also less flexible than static libraries.

This is not clear to me. In which ways are they less flexible?

Reply:
> > Object libraries have their uses, but they are also less flexible than static libraries.
> 
> This is not clear to me. In which ways are they less flexible?

I think, this part of the quote is not related to us as we use CMake >= 3.22.

Reply:
The internal CMake code that handles `STATIC` libraries is older and more robust than code for `OBJECT` ones. Sticking to the former type only allows us to have a predictable behaviour across the whole codebase.